test/e2e/k8s_util.go

```diff
@@ -560,13 +582,17 @@ func (k *KubernetesUtils) waitForPodInNamespace(ns string, pod string) (*string,
 		if k8sPod != nil && k8sPod.Status.Phase == v1.PodRunning {
 			if k8sPod.Status.PodIP == "" {
 				return nil, errors.WithMessagef(err, "unable to get IP of Pod '%s/%s'", ns, pod)
-			} else {
-				log.Debugf("IP of Pod '%s/%s' is: %s", ns, pod, k8sPod.Status.PodIP)
 			}
-
+			var podIPs []string
+			for _, podIP := range k8sPod.Status.PodIPs {
+				podIPs = append(podIPs, podIP.IP)
+			}
+			if len(podIPs) == 0 {
```

Comment:
It checked if it's "" above L577 `k8sPod.Status.PodIP == ""`. When `podIPs` is returned, it must have entry inside it.

Reply:
My bad... did not see there's an extra check up there. Then the question is will there ever be a chance that `Status.PodIP` is not "" and `len(Status.PodIPs)` is 0? I don't think it is possible as I remember some of the controller logic depends on this fact

Reply:
I checked some use case e.g. `./pkg/controller/networkpolicy/endpoint_querier_test.go`. They only set the `PodIP`, so I think it's not mandatory to specify `PodIPs` if `PodIP` is specified. Thus, I add `if len(podIPs) == 0 {...}`.

Reply:
`endpoint_querier_test.go` is a UT file so the spec is made-up. Since this is an E2E test, I think you can rely on the fact podIP and podIPs will both be populated for single stack and dual stack, as stated multiple times in this file https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go#L1566. However, in podStatus conversion functions https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_pods.go#L1566 I do see they account for circumstances where podIP is set and podIPs is not, so I guess we are probably fine here as well

Reply:
Got it.